libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -517,6 +517,109 @@ string YulUtilFunctions::arrayLengthFunction(ArrayType const& _type)
 	});
 }
 
+std::string YulUtilFunctions::resizeDynamicArrayFunction(ArrayType const& _type)
+{
+	solAssert(_type.location() == DataLocation::Storage, "");
+	solAssert(_type.isDynamicallySized(), "");
+	if (!_type.isByteArray() && _type.baseType()->storageBytes() < 32)
+		solAssert(_type.baseType()->isValueType(), "Invalid storage size for non-value type.");
+
+	string functionName = "resize_array_" + _type.identifier();
+	return m_functionCollector->createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(array, newLen) {
+				let oldLen := <fetchLength>(array)
+				<?isByteArray>
+				// For a "long" byte array, store length as 2*length+1
```

Comment:
Hm... I trust you took this from the original implementation... but *why* is this done? It would make sense, if in the other case you'd use ``2*length``, so that you could differentiate between the two cases by the least significant bit of the length, but since that doesn't seem to be the case I wonder what's the point here. Can you comment a link to the original implementation of this for the old codegen for lazy me who is too lazy to look for it myself ;-D?

Reply:
here it is: https://github.com/ethereum/solidity/blob/develop/libsolidity/codegen/ArrayUtils.cpp#L752

Reply:
Mind you, the documentation also mentions this: https://solidity.readthedocs.io/en/latest/miscellaneous.html#bytes-and-string

Reply:
Ok - and the ``*2`` part for non-byte-arrays is in ``<fetchLength>``, resp. in ``YulUtilFunctions:::arrayLengthFunction``? So ``arrayLengthFunction`` decodes the length and here you re-encode it? Is that necessary? Does this work for both short and long bytes arrays and strings?
Maybe I'd need another look at it, but there might be some potential for improvement here...

Reply:
in this PR I kind of skipped the byte-array specific details and concentrated on non-byte non-static storage arrays (I did implement some byte array parts because it seemed easy enough, but whole other parts are missing). The once that works, I want to focus on the other parts.

Reply:
Hm - maybe better to unimplemented-assert the byte array case for now then?

Reply:
it is asserted at other places.

Reply:
I can add it here too if you think that's better?

Reply:
Since the byte array implementation doesn't look complete and/or correct so far: yes, probably better to assert here as well and remove the code parts - then we can implement the byte array case cleanly in one go with tests and all ;-).

Reply:
Alright, will do